src/lib.rs

```diff
@@ -1 +1,452 @@
+
+struct Sgp4;
+
+impl sgp4::Sgp4 for Sgp4 {
+    fn orbit_from_kozai_elements(
+        geopotential: Geopotential,
+        inclination: f64,
+        right_ascension: f64,
+        eccentricity: f64,
+        argument_of_perigee: f64,
+        mean_anomaly: f64,
+        kozai_mean_motion: f64,
+    ) -> Result<Orbit, SgpError> {
+        match original::Orbit::from_kozai_elements(
+            &geopotential.into(),
+            inclination,
+            right_ascension,
+            eccentricity,
+            argument_of_perigee,
+            mean_anomaly,
+            kozai_mean_motion,
+        ) {
+            Ok(orbit) => Ok(orbit.into()),
+            Err(err) => Err(err.into()),
+        }
+    }
+
+    fn wgs72() -> Geopotential {
+        original::WGS72.into()
+    }
+
+    fn wgs84() -> Geopotential {
+        original::WGS72.into()
+    }
+
+    fn afspc_epoch_to_sidereal_time(epoch: f64) -> f64 {
+        original::afspc_epoch_to_sidereal_time(epoch)
+    }
+
+    fn iau_epoch_to_sidereal_time(epoch: f64) -> f64 {
+        original::iau_epoch_to_sidereal_time(epoch)
+    }
+
+    fn parse2les(tles: String) -> Result<Vec<ElementS>, SgpError> {
+        match original::parse_2les(&tles) {
+            Ok(elements) => Ok(elements.into_iter().map(|e| e.into()).collect()),
+            Err(err) => Err(err.into()),
+        }
+    }
+
+    fn parse3les(tles: String) -> Result<Vec<ElementS>, SgpError> {
+        match original::parse_3les(&tles) {
+            Ok(elements) => Ok(elements.into_iter().map(|e| e.into()).collect()),
+            Err(err) => Err(err.into()),
+        }
+    }
+}
+
+struct ResonanceState(Mutex<original::ResonanceState>);
+impl ResonanceState {
+    fn new(state: original::ResonanceState) -> Self {
+        ResonanceState(Mutex::new(state))
+    }
+}
+
+impl sgp4::ResonanceState for ResonanceState {
+    fn t(&self) -> f64 {
+        self.0.lock().expect("The mutex was poisioned").t()
+    }
+}
+
+struct Constants(original::Constants);
+
+impl Constants {
+    fn new_wrap(state: original::Constants) -> Self {
+        Constants(state)
+    }
+}
+
+impl sgp4::Constants for Constants {
+    fn new(
+        geopotential: Geopotential,
+        epoch_to_sidereal_time: EpochToSiderealTimeAlgorithm,
+        epoch: f64,
+        drag_item: f64,
+        orbit0: Orbit,
+    ) -> Result<Handle<Constants>, SgpError> {
+        let epoch_to_sidereal_time_fn = match epoch_to_sidereal_time {
+            EpochToSiderealTimeAlgorithm::Afspc => original::afspc_epoch_to_sidereal_time,
+            EpochToSiderealTimeAlgorithm::Iau => original::iau_epoch_to_sidereal_time,
+        };
+
+        match original::Constants::new(
+            geopotential.into(),
+            epoch_to_sidereal_time_fn,
+            epoch,
+            drag_item,
+            orbit0.into(),
+        ) {
+            Ok(state) => Ok(Handle::new(Constants::new_wrap(state))),
+            Err(err) => Err(err.into()),
+        }
```

Comment:
Because we have an `impl From<original::Error> for SgpError`, we should be able to replace a lot of these `match` statements with `?`.

```suggestion
        let state = original::Constants::new(
            geopotential.into(),
            epoch_to_sidereal_time_fn,
            epoch,
            drag_item,
            orbit0.into(),
        )?;
        
        Ok(Handle::new(Constants::new_wrap(state)))
```